client/src/components/dashboard/DashboardCanvasItem.jsx

```diff
@@ -28,12 +28,33 @@ export default class DashboardCanvasItem extends Component {
     }, 0);
   }
 
-  getItemLayout() {
+  shouldComponentUpdate(nextProps) {
+    const oldLayout = this.getItemLayout(this.props);
+    const newLayout = this.getItemLayout(nextProps);
+    const layoutsExist = Boolean(oldLayout && newLayout);
+    const dimensionsUnchanged = layoutsExist ?
+      oldLayout.w === newLayout.w && oldLayout.h === newLayout.h : false;
+    const canvasWidthUnchanged = this.props.canvasWidth === nextProps.canvasWidth;
+    const needDataset = this.props.item.type === 'visualisation';
+    const datasetDependencyMet = needDataset ? this.getIsDatasetLoaded(this.props) : true;
+    const styleTransitionFinished = this.state.style.boxShadow === 'none';
+
+    const shouldNotUpdate = Boolean(
+        dimensionsUnchanged &&
+        canvasWidthUnchanged &&
+        datasetDependencyMet &&
+        styleTransitionFinished
+      );
+
+    return !shouldNotUpdate;
```

Comment:
the expression `!shoultNotUpdate` reads a bit strange to me as it would be equivalent to `shouldUpdate`?. I wonder if it's possible to rewrite the boolean logic above to avoid the double negation?